lib/orbit_web/controllers/sign_in_controller.ex

```diff
@@ -61,33 +62,51 @@ defmodule OrbitWeb.SignInController do
     )
   end
 
+  @spec validate_override(map()) :: boolean()
+  defp validate_override(override) do
```

Comment:
refactoring suggestion (but probably not): I wish I knew more about ecto changeset validation, because this is the textbook use case for it (taking form input, validating and immediately inserting it to the database, and returning an user-facing error if it's invalid).

If you were motivated to do this in an Ecto-y way, you could try rewriting this to use Ecto validation. But this vanilla elixir validation is perfectly fine and in line with the rest of the app, I've never really understood Ecto's mental model so I can't suggest how to do it, and it's probably not worth the effort to refactor it in an already-complex PR.

Reply:
Ecto.Changeset never even occurred to me here, but I see it too. I know almost nothing about it. This would be good tech debt for us to be forced to learn broadly; I'll make an inbound task

Reply:
https://app.asana.com/0/1200882337457260/1209182581824201/f